#46 Header - The First Version

Pink-Mallow opened the pull request:
## What does this change?

<!-- A PR should have enough detail to be understandable far in the future. e.g what is the problem/why is the change needed, how does it solve it and any questions or points of discussion. Prefer copying information from a GitHub issue over linking to it; the card may not always exist and reviewers may not have access to the board. -->
I have made the Logo component apart from the Header Component and rendered it inside the `<header>` in case the Logo needs to be used separately. 
I have made a basic header that is applied all over the website. Though navigating between pages doesn't work entirely as it somehow doesn't wants to link to the other pages. For now, you can go back to the home page if you face error 404. 


## How Has This Been Tested?
<!-- [Link to test results in the Wiki]()-->
_Link to the issue where i [documented the tests](https://github.com/fdnd-agency/designingforrecognition/issues/46#issuecomment-3452961966)_

- [x]  User test
- [x] Accessibility test
- [x] Performance test
- [x] Responsive Design test
- [x] Device test
- [ ] Browser test

## Images

<!-- Usually only applicable to UI changes, what did it look like before and what will it look like after? -->
- This Image shows the _**desktop**_ version:   
   <img width="2164" height="256" alt="image" src="https://github.com/user-attachments/assets/f1a1075f-13d8-4704-a1e3-84f5bc409035" />

- This image shows the **_Mobile version_**:
   <img width="902" height="140" alt="image" src="https://github.com/user-attachments/assets/5f7d1a56-e939-42fc-9deb-9198caedcdb8" />


## How to review

<!-- Provide instructions to help others verify the change. This could take the form of "On PROD, do X and witness Y. On this branch, do X and witness Z. " -->
- Can you check if i need to make it responsible in size? Is the text too small?
- Do i make the Logo in a colomn way? Or keep it spread out like it shows in the picture above?
- Is the styling well?

Pink-Mallow committed:
feat: Logo Component #53

I have made quick easy version of a logo component. The styling has yet to be made and decided. For now i have made a linear-gradient version of the logo as base.

Pink-Mallow committed:
feat: Header #46

I have yet to make more but for now i have the basic for a header:
- I made a component of the header,
- I put the Logo component inside the header for reuse,
- I made a simple header and have yet to continue work it out for responsiveness and styling.

Pink-Mallow committed:
#46 Styling the header further

I styled the header further and better than how it was before. This time i added Responsiveness to use the hamburger menu.

Next step is making a menu dropdown.

Pink-Mallow committed:
Update Header.svelte

hamburger menu went wrong so i set it on comment for the moment. i'm lost for the moment.

Pink-Mallow committed:
#46 The menu button is finally working

I finally managed to get the menu button working. It is responsive so it only shows on mobile at a certain width.

Pink-Mallow committed:
#46 It is now on all pages available

Pink-Mallow committed:
#46  Small change for Lighthouse test

Pink-Mallow committed:
#46 Just to have something on the home page

vsheo commented on src/routes/+page.svelte:
```diff
@@ -1,153 +1,9 @@
-<h1>Welcome to overview page</h1>
-<p>Visit <a href="https://svelte.dev/docs/kit">svelte.dev/docs/kit</a> to read the documentation</p>
+<Logo />
+<Header />
```
This can be removed, the page.svelte doesn’t need to load these components, since you’re already doing that in +layout.svelte

vsheo reviewed (changes requested):
- Can you check if i need to make it responsible in size? Is the text too small?
  - I think the title should be bigger on desktop, now its at 16px 
- Do i make the Logo in a colomn way? Or keep it spread out like it shows in the picture above?
  - The version in the picture looks good
- Is the styling well?
  - im having problems seeing the mobile version, in your pictures it looks different.
<img width="970" height="719" alt="image" src="https://github.com/user-attachments/assets/9440a66f-ca86-4bfb-acc9-69ae263502d8" />

<img width="756" height="468" alt="image" src="https://github.com/user-attachments/assets/0fd35d00-5172-469f-b388-855211e4b68f" />

- the hamburger menu button animation looks great.

[Menu.webm](https://github.com/user-attachments/assets/0770e323-ab0e-42ec-a082-3e16b28bf56d)

- if you could you also add a transition to the ul so it slides in smoothly, then I think it'll be complete

Pink-Mallow committed:
#46 #80 fix: feedback/review

- removed `<Logo /> <Header />` components from src/+page.svelte as it is unnecessary.
- The title is now responsive as i gave it a clamp().
- Adjusted the Media Queries to make the menu better.

Pink-Mallow committed:
#80 #46  fix: tiny fix in index.js

Pink-Mallow committed:
#80 oops

Pink-Mallow committed:
Merge branch 'dev' into header-46
